compiler/prelude/prelude.go

```diff
@@ -372,7 +372,7 @@ var $internalAppend = function(slice, array, offset, length) {
 
 var $equal = function(a, b, type) {
   if (type === $jsObjectPtr) {
-    return a === b;
+    return a === b || (a.hasOwnProperty("object") && a.object === undefined && b.hasOwnProperty("object") && b.object === undefined);
```

Comment:
Have you reviewed #237? This PR is to fix the issue described there, and it uses the solution I proposed. I should make sure you agree with that proposal first.

> Why was this necessary? Which test was failing?

The test for #237 that I added.

Namely, calling `js.Global.Call("eval", "(foo(null))"))` would print expected output, but calling `js.Global.Call("eval", "(foo(undefined))")` would print:

```Go
	// x: undefined
	// x == nil: false
	// x == js.Undefined: false
```

Instead of:

```Go
	// x: undefined
	// x == nil: false
	// x == js.Undefined: true
```